BUG:import type error in version ^13.0.10 & ^13.0.11

../node_modules/.pnpm/swagger-typescript-api@13.0.10_encoding@0.1.13/node_modules/swagger-typescript-api/index.d.ts:1:39 - error TS2307: Cannot find module './src/schema-parser/mono-schema-parser' or its corresponding type declarations.

1 import type { MonoSchemaParser } from "./src/schema-parser/mono-schema-parser";
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../node_modules/.pnpm/swagger-typescript-api@13.0.10_encoding@0.1.13/node_modules/swagger-typescript-api/index.d.ts:215:46 - error TS2307: Cannot find module './src/translators/translator' or its corresponding type declarations.

215   customTranslator?: new () => typeof import("./src/translators/translator").Translator;
                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../node_modules/.pnpm/swagger-typescript-api@13.0.10_encoding@0.1.13/node_modules/swagger-typescript-api/index.d.ts:282:22 - error TS2307: Cannot find module './src/schema-parser/schema-parser' or its corresponding type declarations.

282       parser: import("./src/schema-parser/schema-parser").SchemaParser,
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../node_modules/.pnpm/swagger-typescript-api@13.0.10_encoding@0.1.13/node_modules/swagger-typescript-api/index.d.ts:369:28 - error TS2307: Cannot find module './src/code-gen-process' or its corresponding type declarations.

369     codeGenProcess: import("./src/code-gen-process").CodeGenProcess,
                               ~~~~~~~~~~~~~~~~~~~~~~~~

../node_modules/.pnpm/swagger-typescript-api@13.0.10_encoding@0.1.13/node_modules/swagger-typescript-api/index.d.ts:660:24 - error TS2307: Cannot find module './src/translators/translator' or its corresponding type declarations.

660     ) => typeof import("./src/translators/translator").Translator;
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../node_modules/.pnpm/swagger-typescript-api@13.0.10_encoding@0.1.13/node_modules/swagger-typescript-api/index.d.ts:664:46 - error TS2307: Cannot find module './src/component-type-name-resolver' or its corresponding type declarations.

664     componentTypeNameResolver: typeof import("./src/component-type-name-resolver").ComponentTypeNameResolver;
                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 6 errors in the same file, starting at: ../node_modules/.pnpm/swagger-typescript-api@13.0.10_encoding@0.1.13/node_modules/swagger-typescript-api/index.d.ts:1

--- a/package.json
+++ b/package.json
@@ -14,26 +14,25 @@
   "exports": {
     ".": {
       "import": {
-        "types": "./index.d.ts",
+        "types": "./dist/types.d.ts",
         "default": "./dist/lib.js"
       },
       "require": {
-        "types": "./index.d.ts",
+        "types": "./dist/types.d.cts",
         "default": "./dist/lib.cjs"
       }
     }
   },
   "main": "./dist/lib.cjs",
   "module": "./dist/lib.js",
-  "types": "./index.d.ts",
+  "types": "./dist/index.d.cts",
   "bin": {
     "sta": "./dist/cli.js",
     "swagger-typescript-api": "./dist/cli.js"
   },
   "files": [
     "dist",
-    "templates",
-    "index.d.ts"
+    "templates"
   ],
   "scripts": {
     "build": "tsup",

--- a/tsup.config.ts
+++ b/tsup.config.ts
@@ -6,6 +6,11 @@ export default defineConfig({
     cli: "index.js",
   },
   clean: true,
+  dts: {
+    entry: {
+      types: "types/index.d.ts",
+    },
+  },
   format: ["esm", "cjs"],
   minify: true,
   sourcemap: true,

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -1,4 +1,4 @@
-import type { MonoSchemaParser } from "./src/schema-parser/mono-schema-parser";
+import type { MonoSchemaParser } from "../src/schema-parser/mono-schema-parser";
 
 type HttpClientType = "axios" | "fetch";
 
@@ -212,7 +212,7 @@ interface GenerateApiParamsBase {
    * }
    * ```
    */
-  customTranslator?: new () => typeof import("./src/translators/translator").Translator;
+  customTranslator?: new () => typeof import("../src/translators/translator").Translator;
   /** fallback name for enum key resolver */
   enumKeyResolverName?: string;
   /** fallback name for type name resolver */
@@ -279,7 +279,7 @@ type PrimitiveTypeStructValue =
   | string
   | ((
       schema: Record<string, any>,
-      parser: import("./src/schema-parser/schema-parser").SchemaParser,
+      parser: import("../src/schema-parser/schema-parser").SchemaParser,
     ) => string);
 
 type PrimitiveTypeStruct = Record<
@@ -366,7 +366,7 @@ export interface Hooks {
   /** Start point of work this tool (after fetching schema) */
   onInit?: <C extends GenerateApiConfiguration["config"]>(
     configuration: C,
-    codeGenProcess: import("./src/code-gen-process").CodeGenProcess,
+    codeGenProcess: import("../src/code-gen-process").CodeGenProcess,
   ) => C | void;
   /** customize configuration object before sending it to ETA templates */
   onPrepareConfig?: <C extends GenerateApiConfiguration>(
@@ -657,11 +657,11 @@ export interface GenerateApiConfiguration {
     /** do not use constructor args, it can break functionality of this property, just send class reference */
     customTranslator?: new (
       ...args: never[]
-    ) => typeof import("./src/translators/translator").Translator;
+    ) => typeof import("../src/translators/translator").Translator;
     internalTemplateOptions: {
       addUtilRequiredKeysType: boolean;
     };
-    componentTypeNameResolver: typeof import("./src/component-type-name-resolver").ComponentTypeNameResolver;
+    componentTypeNameResolver: typeof import("../src/component-type-name-resolver").ComponentTypeNameResolver;
     fileNames: {
       dataContracts: string;
       routeTypes: string;
@@ -754,7 +754,7 @@ export interface GenerateApiOutput {
   renderTemplate: (
     templateContent: string,
     data: Record<string, unknown>,
-    etaOptions?: import("eta/dist/types/config").PartialConfig,
+    etaOptions?: Partial<import("eta").EtaConfig>,
   ) => string;
   getTemplate: (params: {
     fileName?: string;

